src/routes/post_report.js

```diff
@@ -1,28 +1,45 @@
 const connPool = require('../../database/db_connect');
 const dbQueries = require('../db_queries');
+const async = require('../async.js');
 
 module.exports = {
   method: 'POST',
   path: '/post-report',
   handler: (request, reply) => {
-    const postData = JSON.parse(request.payload);
+    const formData = JSON.parse(request.payload);
 
-    dbQueries.postReportDetails(connPool, postData, (err, result) => {
-      if (err) return console.log(err);
+    // insert report details
+    const insertDetails = (postData, callback) => {
+      const insertData = postData;
+      return dbQueries.postReportDetails(connPool, insertData, (postReportDetailsError, result) => {
+        if (postReportDetailsError) return console.log(postReportDetailsError);
 
-      const postId = result.rows[0].post_id;
+        const postId = result.rows[0].post_id;
+        insertData.post_id = postId;
+        return callback(null, insertData);
+      });
+    };
 
-      postData.post_id = postId;
-
-      if (postData.type_tags.length > 0) {
-        return dbQueries.postReportTags(connPool, postData, (postErr) => {
+    // insert posts tags
+    const insertPostsTags = (insertData, callback) => {
+      if (insertData.type_tags.length > 0) {
+        return dbQueries.postReportTags(connPool, insertData, (postErr) => {
           if (postErr) return console.log(postErr);
-          // redirect user to new post view
-          return reply.redirect(`/posts?id=${postId}`);
+          return callback(null, insertData);
         });
       }
+      return callback(null, insertData);
+    };
+    // asynchronously insert details and posts_tags
+    // passing post ID to callback
+    async.waterfall(
+      formData, [insertDetails, insertPostsTags], (error, insertData) => {
+        if (error) return console.log(error);
 
-      return reply.redirect(`/posts?id=${postId}`);
-    });
+        // return post ID to scripts/report.js
+        return reply(JSON.stringify({
```

Comment:
I think you can just reply with insertData.post_id here - not sure whether there are any advantages to creating an object for it?